pulsar/producer.go

```diff
@@ -163,6 +165,19 @@ type ProducerOptions struct {
 	// PartitionsAutoDiscoveryInterval is the time interval for the background process to discover new partitions
 	// Default is 1 minute
 	PartitionsAutoDiscoveryInterval time.Duration
+
+	// EncryptionKeys list of encryption key names to encrypt session key
+	EncryptionKeys []string
```

Comment:
Should we wrap these into a struct? Maybe something like this?
```
type Encryption {
  Keys []string
  KeyReader crypto.KeyReader
  etc
]

```

Reply:
Make sense